server/src/main/java/io/crate/replication/logical/ShardReplicationService.java

```diff
@@ -68,27 +78,74 @@ public void close() throws IOException {
         }
     }
 
+    void getRemoteClusterClient(Index index, Consumer<Client> onSuccess, Consumer<Exception> onFailure) {
+        var subscriptionName = subscriptionName(index.getUUID());
```

Comment:
Minor: I find `subscriptionName` in this context confusing. We have subscriptions which can have a name such as `sub1` but this here is the name of the remote cluster, right ?

Reply:
> but this here is the name of the remote cluster, right ?

No it's not the name of the remote cluster, we register connections based on the subscription name while e.g. OpenSearch register connections based on the remote cluster name.

Reply:
👍